jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -663,6 +722,35 @@ ecma_op_object_put (ecma_object_t *object_p, /**< the object */
       }
       break;
     }
+    case ECMA_OBJECT_TYPE_CLASS:
+    {
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+      /* ES2015 9.4.5.5 */
+      if (ecma_is_typedarray (ecma_make_object_value (object_p)))
+      {
+        ecma_number_t num = ecma_string_to_number (property_name_p);
+        ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
```

Comment:
Hm. so a[5.0] is converted to a[5] during parsing. That makes sense.

And we basically needs the double conversion to return with undefined for "canonical numbers". (ECMA_STRING_CONTAINER_UINT32_IN_DESC handles all normal cases).